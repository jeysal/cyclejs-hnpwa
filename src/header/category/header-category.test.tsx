import { mockDOMSource } from '@cycle/dom';
import o2p from 'observable-to-promise';
import { select } from 'snabbdom-selector';
import xs from 'xstream';

import HeaderCategory from './header-category';

test('renders the basic category list item correctly', async () => {
  const doms = await o2p(
    HeaderCategory('content', '#', false)({
      DOM: mockDOMSource({}),
      active$: xs.of(false),
    }).DOM,
  );

  expect(doms).toMatchSnapshot();
});

test('adds the pull right style if requested', async () => {
  const doms = await o2p(
    HeaderCategory('content', '#', true)({
      DOM: mockDOMSource({}),
      active$: xs.of(false),
    }).DOM,
  );
  expect((doms[0].data as any).props.className).toEqual(
    expect.stringContaining('pullRight'),
  );
});

test('updates the active style', async () => {
  const doms = await o2p(
    HeaderCategory('content', '#', false)({
      DOM: mockDOMSource({}),
      active$: xs.of(false, true),
    }).DOM,
  );

  expect((select('a', doms[0])[0].data as any).props.className).not.toEqual(
    expect.stringContaining('current'),
  );
  expect((select('a', doms[1])[0].data as any).props.className).toEqual(
    expect.stringContaining('current'),
  );
});

test('emits a navigation event on click', async () => {
  const navigations = await o2p(
    HeaderCategory('content', '#', false)({
      DOM: mockDOMSource({ li: { click: xs.of({}) } }),
      active$: xs.of(false),
    }).navigate$,
  );

  expect(navigations).toHaveLength(1);
});
