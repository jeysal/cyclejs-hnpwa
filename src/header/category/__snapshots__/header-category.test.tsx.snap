// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the basic category list item correctly 1`] = `
Array [
  <li
    className="item "
  >
    <a
      href="#"
      className="link "
    >
      content
    </a>
  </li>,
]
`;
